### PR TITLE
speed: Document the deficiencies of the command

### LINKED ISSUE #7032: speed rejects most digest/cipher algorithms as unknown
Contrary to what the _speed_ manpage suggests, the _openssl speed_ app does not take all the digests/ciphers as outputted by _list -digest-commands_ and _list -cipher-commands_. Those algorithms which do not work if given explicitly seem also to be omitted when no algorithm option is given, and the "pre-compiled grand selection is tested".

Digests supported by _speed_ are e.g. md4, md5, sha1, sha256

Verified with recent master (OpenSSL 1.1.1-pre10-dev) and 1.1.0f

Issue #6922 appears to be somewhat related, while the EdDSA signatures (even the working ecdhx25519) mentioned there might not be neither referenced explicitly by the speed man page nor do they seem to be retrievable by any of the mentioned _list_ options.

To reproduce, in Bash, do:
$ for alg in \`openssl list -digest-commands\`; do openssl speed $alg 2>&1 | grep Unknown; done
speed: Unknown algorithm blake2b512
speed: Unknown algorithm blake2s256
speed: Unknown algorithm gost
speed: Unknown algorithm sha224
speed: Unknown algorithm sha3-224
speed: Unknown algorithm sha3-256
speed: Unknown algorithm sha3-384
speed: Unknown algorithm sha3-512
speed: Unknown algorithm sha384
speed: Unknown algorithm sha512-224
speed: Unknown algorithm sha512-256
speed: Unknown algorithm shake128
speed: Unknown algorithm shake256
speed: Unknown algorithm sm3

$ for alg in \`openssl list -cipher-commands\`; do openssl speed $alg 2>&1 | grep Unknown; done
speed: Unknown algorithm aes-128-ecb
speed: Unknown algorithm aes-192-ecb
speed: Unknown algorithm aes-256-ecb
speed: Unknown algorithm aria-128-cbc
speed: Unknown algorithm aria-128-cfb
speed: Unknown algorithm aria-128-cfb1
speed: Unknown algorithm aria-128-cfb8
speed: Unknown algorithm aria-128-ctr
speed: Unknown algorithm aria-128-ecb
speed: Unknown algorithm aria-128-ofb
speed: Unknown algorithm aria-192-cbc
speed: Unknown algorithm aria-192-cfb
speed: Unknown algorithm aria-192-cfb1
speed: Unknown algorithm aria-192-cfb8
speed: Unknown algorithm aria-192-ctr
speed: Unknown algorithm aria-192-ecb
speed: Unknown algorithm aria-192-ofb
speed: Unknown algorithm aria-256-cbc
speed: Unknown algorithm aria-256-cfb
speed: Unknown algorithm aria-256-cfb1
speed: Unknown algorithm aria-256-cfb8
speed: Unknown algorithm aria-256-ctr
speed: Unknown algorithm aria-256-ecb
speed: Unknown algorithm aria-256-ofb
speed: Unknown algorithm base64
speed: Unknown algorithm bf-cfb
speed: Unknown algorithm bf-ecb
speed: Unknown algorithm bf-ofb
speed: Unknown algorithm camellia-128-cbc
speed: Unknown algorithm camellia-128-ecb
speed: Unknown algorithm camellia-192-cbc
speed: Unknown algorithm camellia-192-ecb
speed: Unknown algorithm camellia-256-cbc
speed: Unknown algorithm camellia-256-ecb
speed: Unknown algorithm cast5-cbc
speed: Unknown algorithm cast5-cfb
speed: Unknown algorithm cast5-ecb
speed: Unknown algorithm cast5-ofb
speed: Unknown algorithm des-cfb
speed: Unknown algorithm des-ecb
speed: Unknown algorithm des-ede
speed: Unknown algorithm des-ede-cbc
speed: Unknown algorithm des-ede-cfb
speed: Unknown algorithm des-ede-ofb
speed: Unknown algorithm des-ede3-cbc
speed: Unknown algorithm des-ede3-cfb
speed: Unknown algorithm des-ede3-ofb
speed: Unknown algorithm des-ofb
speed: Unknown algorithm des3
speed: Unknown algorithm desx
speed: Unknown algorithm idea-cfb
speed: Unknown algorithm idea-ecb
speed: Unknown algorithm idea-ofb
speed: Unknown algorithm rc2-40-cbc
speed: Unknown algorithm rc2-64-cbc
speed: Unknown algorithm rc2-cfb
speed: Unknown algorithm rc2-ecb
speed: Unknown algorithm rc2-ofb
speed: Unknown algorithm rc4-40
speed: Unknown algorithm seed-cfb
speed: Unknown algorithm seed-ecb
speed: Unknown algorithm seed-ofb
speed: Unknown algorithm sm4-cbc
speed: Unknown algorithm sm4-cfb
speed: Unknown algorithm sm4-ctr
speed: Unknown algorithm sm4-ecb
speed: Unknown algorithm sm4-ofb

### PR DESCRIPTION
**Pull request by t8m**
Fixes #7032

##### Checklist

- [x] documentation is added or updated

### CODE REVIEW

File: doc/man1/openssl-speed.pod.in [hunk 1]
```diff
@@ -112,9 +111,18 @@ pre-compiled grand selection is tested.
 
 =back
 
+=head1 BUGS
+
+The I<algorithm> can be selected only from a pre-compiled subset of things
+that the C<openssl speed> command knows about. To test any additional digest
+or cipher algorithm supported by OpenSSL use the C<-evp> option.
+
+There is no way to test speed of additional public key algorithms supported
```
Comment (paulidale) [inline doc/man1/openssl-speed.pod.in:120]: _... to test **the** speed of **any** additional ..._

**Review by paulidale — APPROVED** [review]
One wording change.  Approved once that's done.

File: doc/man1/openssl-speed.pod.in (hunk 1 above)
Comment (t8m) [reply doc/man1/openssl-speed.pod.in]: Fixed.

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
The last fixup just drops one incorrect hunk of the original patch. So I am assuming the approval holds.

**Review by paulidale — APPROVED** [review]

**Pull request closed without merging**